src/services/EcoBridge/OmniBridge/OmniBridge.utils.ts

```diff
@@ -0,0 +1,824 @@
+  const toEthersProvider = providers[toChainId]
+
+  if (address === ADDRESS_ZERO && mode === 'NATIVE') {
+    const toAddress = '0x6A023CCd1ff6F2045C3309768eAd9E68F978f6e1'
+    return fetchTokenDetails(
+      bridgeDirection,
+      {
+        address: toAddress,
+        chainId: toChainId,
+      },
+      toEthersProvider
+    )
+  }
+
+  if (!fromMediatorAddress || !toMediatorAddress) return
+
+  const abi = [
+    'function isRegisteredAsNativeToken(address) view returns (bool)',
+    'function bridgedTokenAddress(address) view returns (address)',
+    'function nativeTokenAddress(address) view returns (address)',
+  ]
+  const fromMediatorContract = new Contract(fromMediatorAddress, abi, fromEthersProvider)
+  const isNativeToken = await fromMediatorContract.isRegisteredAsNativeToken(address)
+
+  if (isNativeToken) {
+    const toMediatorContract = new Contract(toMediatorAddress, abi, toEthersProvider)
+
+    const toAddress = await toMediatorContract.bridgedTokenAddress(address)
+
+    const toName = await getToName(fromToken, toChainId, toAddress, toEthersProvider)
+
+    const abiDecimals = ['function decimals() view returns(uint)']
+    const decimals = await new Contract(toAddress, abiDecimals, toEthersProvider).decimals()
+
+    return {
+      name: toName,
+      chainId: toChainId,
+      address: toAddress,
+      mode: 'erc677',
+      mediator: toMediatorAddress,
+      decimals: decimals.toString(),
+    }
+  }
+
+  const toAddress = await fromMediatorContract.nativeTokenAddress(address)
+
+  const toName = await getToName(fromToken, toChainId, toAddress, toEthersProvider)
+
+  const abiDecimals = ['function decimals() view returns(uint)']
+  const decimals = await new Contract(toAddress, abiDecimals, toEthersProvider).decimals()
+
+  return {
+    name: toName,
+    chainId: toChainId,
+    address: toAddress,
+    mode: 'erc20',
+    mediator: toMediatorAddress,
+    decimals: decimals.toString(),
+  }
+}
+
+export const fetchToToken = async (
+  bridgeDirection: string,
+  fromToken: TokenWithAddressAndChain & { mode: string; name: string },
+  toChainId: ChainId,
+  providers?: EcoBridgeProviders
+) => {
+  try {
+    return await fetchToTokenDetails(bridgeDirection, fromToken, toChainId, providers)
+  } catch (e) {
+    return
+  }
+}
+
+//calculate fee and toAmount
+const processMediatorData = async (
+  direction: string,
+  provider?: Provider
+): Promise<{ feeManagerAddress: string; currentDay: BigNumber } | undefined> => {
+  const abi = [
+    'function getCurrentDay() view returns (uint256)',
+    'function feeManager() public view returns (address)',
+    'function getBridgeInterfacesVersion() external pure returns (uint64, uint64, uint64)',
+  ]
+
+  const { homeMediatorAddress } = BRIDGE_CONFIG[direction]
+
+  const mediatorContract = new Contract(homeMediatorAddress, abi, provider)
+
+  const [interfaceVersion, currentDay] = await Promise.all<BigNumber[], BigNumber>([
```

Comment:
Tuple will work here
```suggestion
  const [interfaceVersion, currentDay] = await Promise.all<[BigNumber[], BigNumber]>([
```

Reply:
fixed [a2c2761](https://github.com/levelkdev/swapr-dapp/pull/1007/commits/a2c2761ff4b65a4f7e3d129ecd0d7e484177d0b0)